### ISSUE TITLE
Image segmentation (of the spinal cord/CSF) with missing modalities

### ISSUE THREAD

**Issue by leandervaneekelen** [ivadomed/ivadomed#686]
**Context**
We often train multi-modal deep learning models (in our case, various contrasts of MRI, e.g. T1(ce), T2, T2*, FLAIR, etc) to take advantage of the complementary information that these modalities give about the underlying anatomy. However, in a realistic setting, each subject in a dataset might have a variable number of available modalities - after all, sometimes data gets corrupted, a patient misses a scan, data goes missing. Traditional deep learning models cannot handle this scenario of missing modalities well. 

Developing models that would be able to segment structures in images with an arbitrary combination of modalities would be a great boon to the overall applicability of deep learning to neural imaging and clinical practice.

**Existing work**
Ivadomed currently has an implementation of the HeMIS framework (Hetero-modal image segmentation, originally proposed by [Havaei et al](https://arxiv.org/abs/1607.05194)). In brief, HeMIS is a convolutional neural network (CNN) architecture that works by having an independent convolutional branch per modality; all branches then feed into a single latent space by calculating statistics over each branch. The latent space contains 'information' from all modalities and can then be further processed to get the desired segmentation. If a modality is missing, that branch is simply skipped and won't contribute to the latent space. See the figure below.

![image](https://user-images.githubusercontent.com/47320151/107994263-96e4d700-6fdc-11eb-82bf-93fb2506850a.png)

I (Leander van Eekelen) have done a short internship in the group on applying HeMIS for spinal cord and CSF segmentation in MRI (various contrasts) and have gotten slightly lackluster results. The details about the done work can be found in the repository dedicated to the project [here](https://github.com/ivadomed/pipeline-hemis).

For the continuation of the project, we want to consider other (more state of the art) alternatives. 

**Potential future avenues**
One promising avenue of research is teacher-student networks for knowledge distillation. We train a teacher network for each available modality and have all of these teachers integrate their knowledge into a single student network. Recommended reading material is, among others:
* https://dl.acm.org/doi/abs/10.1145/3394486.3403234
* https://ieeexplore.ieee.org/document/8979396
* https://arxiv.org/abs/1908.06683

**Comment by arbalest2** [comment]
This paper looks promising :

[SMIL: Multimodal Learning with Severely Missing Modality](https://www.aaai.org/AAAI21Papers/AAAI-437.MaM.pdf)
[paper talk](https://slideslive.com/38947797/smil-multimodal-learning-with-severely-missing-modality)

- Their approach can deal with missing modalities in the training and testing set (works well even if 90% of the training data contains missing modality). I assume even in MRI data modalities can be missing from both train and test
- They use a Meta learning based approach that builds upon [MAML](https://arxiv.org/abs/1703.03400) and achieve SOTA results on three benchmarks.
- Although they use classification models, we can modify the approach and model architecture for image segmentation.
